include/maths/CNaiveBayes.h

```diff
@@ -222,30 +226,47 @@ class MATHS_EXPORT CNaiveBayes {
     std::string print() const;
 
 private:
-    using TFeatureDensityPtr = std::shared_ptr<CNaiveBayesFeatureDensity>;
+    using TFeatureDensityPtr = std::unique_ptr<CNaiveBayesFeatureDensity>;
     using TFeatureDensityPtrVec = std::vector<TFeatureDensityPtr>;
 
     //! \brief The data associated with a class.
-    struct SClass {
+    class CClass {
+    public:
+        CClass() = default;
+        explicit CClass(double count);
+        CClass(const CClass& other);
+
         //! Initialize by reading state from \p traverser.
         bool acceptRestoreTraverser(const SDistributionRestoreParams& params,
                                     core::CStateRestoreTraverser& traverser);
         //! Persist state by passing information to \p inserter.
         void acceptPersistInserter(core::CStatePersistInserter& inserter) const;
+
+        //! Get the number of examples in this class.
+        double count() const;
+        //! Set the number of examples in this class.
```

Comment:
nit: It's not really a setter but exposes count for writes, maybe just "access to number of ... "

Reply:
True. I started off making this a setter, but then decided to make it return a writable reference.